lms/validation/_base.py

```diff
@@ -82,12 +80,15 @@ def parse(self, *args, **kwargs):
 
         :raise lms.validation.ValidationError: if the request isn't valid
         """
-        kwargs.setdefault("locations", self.locations)
-        return self._parser.parse(self, self.context["request"], *args, **kwargs)
+
+        parser = pyramidparser.PyramidParser(
+            location=kwargs.pop("location", self.location),
+            error_handler=self._handle_error,
+        )
+        return parser.parse(self, self.context["request"], *args, **kwargs)
 
     @staticmethod
-    @_parser.error_handler
-    def _handle_error(error, _req, _schema, _status_code, _headers):
+    def _handle_error(error, _req, _schema, *, error_status_code, error_headers):
```

Comment:
The names of the `status_code` and `headers` args changed to `error_status_code` and `error_headers` and they became keyword-only arguments.

Also, we can no longer use `@_parser.error_handler` because `_parser` is no longer a class-level attr. We had to pass this `_handle_error` method to `pyramidparser.PyramidParser()` above instead.